lib/baby_squeel/join_dependency.rb

```diff
@@ -119,7 +119,7 @@ def build(relation, buckets)
 
         join_list = join_nodes + joins
 
-        alias_tracker = Associations::AliasTracker.create(relation.klass.connection, relation.table.name, join_list)
+        alias_tracker = Associations::AliasTracker.create(BabySqueel::ActiveRecord::VersionHelper.at_least_7_2? ? relation.klass.connection.pool : relation.klass.connection, relation.table.name, join_list)
```

Comment:
Can you please point to the Documentation why this change is needed.

Reply:
the first param for the init method for `Associations::AliasTracker` was changed to accept a `pool` now: https://github.com/rails/rails/blob/v7.2.1.2/activerecord/lib/active_record/associations/alias_tracker.rb#L9